File: examples/response.php
<?php
declare(strict_types=1);

/**
 * Generate a standardized JSON response.
 *
 * @param array|null $data   Response payload
 * @param int        $status HTTP-like status code (100-599)
 * @param string     $message Human-readable status message
 *
 * @return string JSON encoded response
 *
 * @throws InvalidArgumentException When inputs are invalid
 * @throws RuntimeException         When encoding fails
 */
function jsonResponse($data = [], $status = 200, $message = ''): string
{
    if (!is_int($status) || $status < 100 || $status > 599) {
        throw new InvalidArgumentException('Status must be a valid HTTP status code (100-599)');
    }

    if (!is_string($message)) {
        throw new InvalidArgumentException('Message must be a string');
    }

    if ($data === null) {
        $data = [];
    }

    if (!is_array($data)) {
        throw new InvalidArgumentException('Data must be an array or null');
    }

    $response = [
        'status' => $status,
        'message' => $message,
        'data' => $data,
    ];

    $encoded = json_encode($response, JSON_PRETTY_PRINT);

    if ($encoded === false) {
        throw new RuntimeException('Failed to encode response as JSON');
    }

    return $encoded;
}

// Example
if (basename(__FILE__) === basename((string) ($_SERVER['SCRIPT_FILENAME'] ?? ''))) {
    echo jsonResponse(['user' => 'John'], 200, 'Success') . PHP_EOL;
}

File: examples/response.go
// Package examples provides a standard JSON response format
package examples

import (
    "encoding/json"
    "fmt"
)

// Response defines the structure of the JSON output.
type Response struct {
    Status  int         `json:"status"`
    Message string      `json:"message"`
    Data    interface{} `json:"data"`
}

// JsonResponse creates a formatted JSON response string with zero dependencies.
// It validates the inputs and returns a JSON string or an error.
func JsonResponse(data interface{}, status int, message string) (string, error) {
    if status < 100 || status > 599 {
        return "", fmt.Errorf("status must be a valid HTTP status code (100-599)")
    }

    response := Response{
        Status:  status,
        Message: message,
        Data:    data,
    }

    jsonString, err := json.MarshalIndent(response, "", "  ")
    if err != nil {
        return "", fmt.Errorf("failed to marshal JSON: %w", err)
    }

    return string(jsonString), nil
}
